dev/com.ibm.ws.security.audit.source/src/com/ibm/ws/security/audit/encryption/AuditCrypto.java

```diff
@@ -1872,14 +1928,20 @@ static final byte[] generateSecretKey(int length) {
         return new SecureRandom().getSeed(length);
     }
 
-    static final byte[] generate3DESKey() {
+    static final byte[] generateSharedKey() {
         byte[] rndSeed = null;
         try {
-            KeyGenerator keyGen = null;
-            keyGen = KeyGenerator.getInstance(ENCRYPT_ALGORITHM);
-            keyGen.init(new SecureRandom());
-            SecretKey key = keyGen.generateKey();
-            rndSeed = key.getEncoded();
+            if (fips140_3Enabled) {
+                int len = 32;
+                rndSeed = new byte[len];
+                random(rndSeed, 0, len);
```

Comment:
Why for FIPS 140-3 do we do it this way instead of using a KeyGenerator?  KeyGenerator we would use doesn't support encoding?

Reply:
The LTPA code uses the random method. We wants the audit code to be consistent with the LTPA code.